vod/manifest_utils.c

```diff
@@ -48,12 +48,41 @@ manifest_utils_write_bitmask(u_char* p, uint32_t bitmask, u_char letter)
 	return p;
 }
 
-static uint32_t*
+static u_char*
+manifest_utils_write_bitmask(u_char* p, uint64_t* bitmask, uint64_t* mask_temp, uint32_t max_bits, u_char letter)
+{
+	uint32_t i;
+
+	vod_memcpy(mask_temp, bitmask, sizeof(bitmask[0]) * vod_array_length_for_bits(max_bits));
+
+	i = vod_get_lowest_bit_set(mask_temp, max_bits);
```

Comment:
i'd prefer to avoid the two lines that call vod_get_lowest_bit_set, and use -
```
for (;;)
{
    i = vod_get_lowest_bit_set(mask_temp, max_bits);
    if (i < 0)
    {
        break;
    }
    ...
```

Reply:
also, i think there's a bug here - with the while impl, it should be >= 0, not > 0